### DISCUSSION TITLE
Having issue getting count from nested query using MongoDB

### DISCUSSION THREAD

**Discussion by jungraejang** [prisma/prisma#21556] [Q&A] [answered]
### Question

I have my Prisma schema setup as following:

```prisma
model Legislation {
    id                        String       @id @map("_id")
    billId                    Int?         @unique @map("id")
    label                    String?
    title                     String
    {......}
    updated            Boolean
    aiMetadata        AiMetadata?
}
```

```ts
type AiMetadata {
    title              String
    summary            String
    categories         String[] 
    moderationFlag     Boolean
    moderationAnalysis String
}
```

I am using tRPC along with Prisma to generate query that looks like this:

```ts
const updatedItemsCount = await prisma.legislation.count({
        where: {
          aiMetadata: {
            moderationFlag: true,
          },
          updated: true,
        },
      });
```

The above code is giving me errors which are the following:

```
Type '{ moderationFlag: true; }' is not assignable to type '(Without<HrfMetadataNullableCompositeFilter, HrfMetadataObjectEqualityInput> & HrfMetadataObjectEqualityInput) | (Without<...> & HrfMetadataNullableCompositeFilter) | null | undefined'.

Type '{ moderationFlag: true; }' is not assignable to type 'Without<HrfMetadataNullableCompositeFilter, HrfMetadataObjectEqualityInput> & HrfMetadataObjectEqualityInput'.

 Type '{ moderationFlag: true; }' is missing the following properties from type 'HrfMetadataObjectEqualityInput': title, summary, moderationAnalysists(2322)
```

Can anybody help me create query that works? All my other queries are working fine but this nested one is giving me some trouble. 

### Information about Prisma Schema, Client Queries and Environment (optional)

```prisma
// Add your schema.prisma
```

```ts
// Add any relevant Prisma Client queries here
```

- OS: 
- Database: 
- Node.js version: 

- Run `prisma -v` to see your Prisma version and paste it

**Comment by ludralph** [comment, marked as answer]
Hi @jungraejang 👋 

Thank you for raising this question.

You're using a [composite type](https://www.prisma.io/docs/concepts/components/prisma-client/composite-types#considerations-when-using-composite-types) and the method `count` does not support composite operations. Prisma does not currently support filtering on fields of composite types directly in the where clause of a query. Composite types in Prisma are similar to embedded documents in MongoDB and are used to embed records inside other records.

So, in your case, you would not be able to filter on the `moderationFlag` field of the `AiMetadata` composite type directly in the where clause of your query. You may need to change the composite type to a model if you would need to frequently perform this operation.

If this answers your question, it would be great if you could **mark this Discussion as answered** to indicate that it has been resolved. 

Otherwise please let us know how else we can help you further or close the Discussion if it was resolved in some other way 🙏

**Reply by jungraejang** [reply]
Thank you for the answer! I guess I will just use raw query for now then.
